src/openms/include/OpenMS/ANALYSIS/XLMS/OpenProXLUtils.h

```diff
@@ -0,0 +1,302 @@
+
+      ProteinProteinCrossLinkType getType() const
+      {
+        if (!beta.empty()) return CROSS;
+
+        if (cross_link_position.second == -1) return MONO;
+
+        return LOOP;
+      }
+
+    };
+
+struct CrossLinkSpectrumMatch
+  {
+    /// structure of the cross-link
+    ProteinProteinCrossLink cross_link;
+
+    /// reference to pair of spectra
+    Size scan_index_light;
+    Size scan_index_heavy;
+
+    /// final score
+    double score;
+
+    /// rank among the matches to the same spectrum
+    Size rank;
+
+    /// counts, scores and other data for xQuest-like output
+    double pre_score;
+    double percTIC;
+    double wTIC;
+    double int_sum;
+    double match_odds;
+    std::vector< double > xcorrx;
+    double xcorrx_max;
+    std::vector< double > xcorrc;
+    double xcorrc_max;
+    Size matched_common_alpha;
+    Size matched_common_beta;
+    Size matched_xlink_alpha;
+    Size matched_xlink_beta;
+    double HyperCommon;
+    double HyperXlink;
+    double HyperAlpha;
+    double HyperBeta;
+    double HyperBoth;
+    double PScoreCommon;
+    double PScoreXlink;
+    double PScoreAlpha;
+    double PScoreBeta;
+    double PScoreBoth;
+
+    std::vector<PeptideHit::FragmentAnnotation> frag_annotations;
+
+    Size peptide_id_index;
+//    PeptideIdentification *peptide_id = NULL;
+
+    bool operator<(const CrossLinkSpectrumMatch& other) const
+    {
+      return score < other.score;
+    }
+
+    bool operator==(const CrossLinkSpectrumMatch& other) const
+    {
+      return cross_link.alpha == other.cross_link.alpha &&
+           cross_link.beta == other.cross_link.beta &&
+           cross_link.cross_link_position == other.cross_link.cross_link_position &&
+           score == other.score &&
+           pre_score == other.pre_score &&
+           percTIC == other.percTIC &&
+           wTIC == other.wTIC &&
+           int_sum == other.int_sum &&
+           match_odds == other.match_odds &&
+           xcorrx == other.xcorrx &&
+           xcorrx_max == other.xcorrx_max &&
+           xcorrc == other.xcorrc &&
+           xcorrc_max == other.xcorrc_max &&
+           matched_common_alpha == other.matched_common_alpha &&
+           matched_common_beta == other.matched_common_beta &&
+           matched_xlink_alpha == other.matched_xlink_alpha &&
+           matched_xlink_beta == other.matched_xlink_beta &&
+           rank == other.rank;
+    }
+
+  };
+
+
+  struct OPENMS_DLLAPI OpenProXLUtils
+  {
+
+    struct XLPrecursor
+    {
+      double precursor_mass;
+      Size alpha_index;
+      Size beta_index;
+
+      bool operator<(const XLPrecursor& other) const
+      {
+        return precursor_mass < other.precursor_mass;
+      }
+
+      bool operator<(const double other) const
+      {
+        return precursor_mass < other;
+      }
+  };
+
+  enum PeptidePosition
+  {
+    INTERNAL = 0,
+    C_TERM = 1,
+    N_TERM = 2
+  };
+
+  struct PeptideMass
```

Comment:
might be a misnomer as the class also contains the relatively heavy AASequence object.
Also consider providing predicates instead of the second < operator